File: plat/js/entities/flamethrower.js
import * as Assets from 'assets';
import Entity from '../entity.js';
import CollisionType from '../collisiontype.js';
import { World } from '../game.js';
class Flamethrower extends Entity {
    type = 'Flamethrower';
    sprite = Assets.find('flamethrower');
    collidable = CollisionType.Enabled;
    startOn;
    spawned = false;
    constructor(args) {
        super(args);
        this.startOn = args.properties?.StartOn ?? false;
    }
    preupdate(ticks, dt) {
        if (this.spawned)
            return;
        this.spawned = true;
        const ent = World().spawnEntity('Flame');
        ent.startOn = this.startOn;
        ent.pos = [...this.pos];
        ent.pos[1] -= ent.size[1];
    }
    die() { }
}
export { Flamethrower };

File: csgame/entities/Flamethrower.cs
using Slate2D;

[Spawnable]
class Flamethrower : Entity
{
    bool StartOn = false;
    bool Spawned = false;

    public Flamethrower(LDTKEntity ent) : base(ent)
    {
        Collidable = CollisionType.Enabled;
        Sprite = Assets.Find("flamethrower");

        StartOn = ent.Properties?["StartOn"]?.Bool ?? false;
    }

    public override void PreUpdate(uint ticks, float dt)
    {
        if (Spawned) return;

        Spawned = true;
        var ent = Main.World.SpawnEntity("Flame") as Flame;
        if (ent == null) return;

        ent.StartOn = StartOn;
        ent.Pos = Pos;
        ent.Pos.Y -= ent.Size.H;
    }
}